### PR TITLE
Migrate e2e tests to KIND

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/4606

We are moving away from appr but we'll need apprclient for a while. So this re-enables the e2e tests, migrates it to KIND and removes the usage of e2e-harness.

Code is a bit outdated so this is "make it work" mode.

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -36,6 +73,26 @@
   pruneopts = "UT"
   revision = "4dd4911251929a0ac4c4f3d8c4bda482a9e09f30"
 
+[[projects]]
+  branch = "master"
+  digest = "1:9c94a9509b279c5d406688d39bcd86f59e66c91e6c54081501dce4ce664383c8"
+  name = "github.com/giantswarm/e2e-harness"
```
Comment (sslavic) [inline Gopkg.lock]: Please vendor deps once more. When I run `dep ensure -update` this one and few other deps get actually removed.

Comment (rossf7) [reply Gopkg.lock]: @sslavic Thanks done. PTAL

**Comment by sslavic** [comment]
Thanks, this fixes https://github.com/giantswarm/giantswarm/issues/7069

**Review by sslavic — APPROVED** [review]
LGTM

**Pull request merged**